ui/src/app/applications/components/applications-list/applications-table.tsx

```diff
@@ -40,7 +40,9 @@ export const ApplicationsTable = (props: {
                             key={app.metadata.name}
                             className={`argo-table-list__row
                 applications-list__entry applications-list__entry--health-${app.status.health.status} ${selectedApp === i ? 'applications-tiles__selected' : ''}`}>
-                            <div className={`row applications-list__table-row`} onClick={e => ctx.navigation.goto(`/applications/${app.metadata.name}`, {}, {event: e})}>
+                            <div
+                                className={`row applications-list__table-row`}
+                                onClick={e => ctx.navigation.goto(`/applications/${app.metadata.name}`, {}, {event: e, replace: true})}>
```

Comment:
I think the ApplicationList -> App Details page navigation should be recorded in history. Please revert this line change.